plumbing/strgdls/store.go

```diff
@@ -58,3 +44,45 @@ func (store *Store) Put(storageDeal *storagedeal.Deal) error {
 
 	return nil
 }
+
+// Has checks if a deal is already present in the datastore corresponding to
+// given proposal
+// bool should be ignored in case of a non-nil error
+func (store *Store) Has(storageProposal *storagedeal.Proposal) (bool, error) {
+	key := datastore.KeyWithNamespaces([]string{StorageDealPrefix, storageProposal.PieceRef.String()})
+	exists, err := store.dealsDs.Has(key)
+	if err != nil || exists == false {
+		return false, err
+	}
+
+	deals, err := store.lsWithPrefix("/" + key.String())
+	if err != nil {
+		return false, err
+	}
+
+	for _, d := range deals {
+		if d.Miner == storageProposal.MinerAddress {
+			return true, nil
+		}
+	}
+
+	return false, nil
+}
+
+func (store *Store) lsWithPrefix(prefix string) ([]*storagedeal.Deal, error) {
```

Comment:
oof... this is going to conflict with my work in #2749

we don't need to worry about it in this PR, but we're probably going to have a nasty rebase to do later